packages/jest-matchers/src/index.js

```diff
@@ -10,27 +10,37 @@
 
 'use strict';
 
-import type {ThrowingMatcherFn, ExpectationResult} from '../types';
+import type {
+  Expect,
+  ExpectationResult,
+  ExpectationObject,
+  MatchersObject,
+  RawMatcherFn,
+  ThrowingMatcherFn,
+} from '../types';
 
 const matchers = require('./matchers');
+const GLOBAL_MATCHERS_OBJECT_KEY = '_JEST_MATCHERS';
+global[GLOBAL_MATCHERS_OBJECT_KEY] = {};
```

Comment:
this will not work until https://github.com/facebook/jest/pull/1275 is merged.
because right now jest-matchers is executed in outside environment